### PR TITLE
scheduler: low space tikv can not balance 

### PR DESCRIPTION
**Pull request by bufferflies**
Signed-off-by: bufferflies <1045931706@qq.com>

### What problem does this PR solve?
close: #4323

### What is changed and how it works?
available space unit is not same 
### Check List

Tests

- Unit test
- Integration test

capacity
  store1,store4: 161gb
  others: 215gb
available ratio and score
image

balance:
image

there is no balance roll back 

Code changes

Side effects

Related changes

### Release note

```release-note
None
```

### CODE REVIEW

**Review by lhy1024 — COMMENTED** [review]
I think we should add more unit test to cover all params in this function.

File: server/core/store.go [hunk 1]
```diff
@@ -346,6 +346,7 @@ func (s *StoreInfo) regionScoreV2(delta int64, lowSpaceRatio float64) float64 {
 	A := float64(s.GetAvgAvailable()) / gb
 	C := float64(s.GetCapacity()) / gb
 	R := float64(s.GetRegionSize() + delta)
+	log.Info("score v2", zap.Float64("A", A), zap.Float64("C", C), zap.Float64("R", R))
```
Comment (lhy1024) [inline server/core/store.go:349]: it is unnecessary

**Review by nolouch — APPROVED** [review]

File: server/core/test_util.go [hunk 2]
```diff
@@ -84,6 +84,25 @@ func NewTestRegionInfo(start, end []byte) *RegionInfo {
 	}}
 }
 
+// NewStoreInfoWithAvailable is create with store
+func NewStoreInfoWithAvailable(id, available, capacity uint64) *StoreInfo {
+	amp := uint64(2)
```
Comment (lhy1024) [inline server/core/test_util.go:89]: why is 2?

**Review by HunDunDM — APPROVED** [review]

File: server/core/test_util.go [hunk 3]
```diff
@@ -84,6 +84,25 @@ func NewTestRegionInfo(start, end []byte) *RegionInfo {
...
+	stats := &pdpb.StoreStats{}
+	stats.Capacity = capacity
+	stats.Available = available
+	usedSize := capacity - available
+	regionSize := (usedSize * amp) * 1000 / gb
```
Comment (HunDunDM) [inline server/core/test_util.go:94]: `mb`?

**Review by HunDunDM — CHANGES_REQUESTED** [review]

File: server/core/test_util.go (hunk 2 above)
Comment (lhy1024) [reply server/core/test_util.go:89]: if only for the test, I think it is better to add the comment "for test".

Comment (lhy1024) [reply server/core/test_util.go:89]: Maybe to add more different amp?

File: server/core/test_util.go (hunk 3 above)
Comment (bufferflies) [reply server/core/test_util.go:94]: ye,region size‘s unit is mb.

**Review by lhy1024 — COMMENTED** [review]
LGTM，PTAL @HunDunDM

File: server/core/store_test.go [hunk 4]
```diff
@@ -127,3 +127,47 @@ func (s *testStoreSuite) TestLowSpaceRatio(c *C) {
...
+		// but the bigger's amp is bigger
+		bigger: NewStoreInfoWithAvailable(1, 10*gb, 100*gb, 1.5),
+		small:  NewStoreInfoWithAvailable(2, 10*gb, 100*gb, 1.4),
+	}, {
+		// store1 and store2 has same  capacity and regionSize（40g），
+		// but store1 has less available space size
+		bigger: NewStoreInfoWithAvailable(1, 60*gb, 100*gb, 1),
+		small:  NewStoreInfoWithAvailable(2, 80*gb, 100*gb, 2),
+	}, {
+		// store1 and store2 has same  capacity and store2(40g) has twice usedSize than store1(20g)
```
Comment (rleungx) [inline server/core/store_test.go:163]:
```suggestion
		// store1 and store2 has same capacity and store2 (40g) has twice usedSize than store1 (20g)
```

File: server/core/store_test.go [hunk 5]
```diff
@@ -127,3 +127,47 @@ func (s *testStoreSuite) TestLowSpaceRatio(c *C) {
...
+		// but the store1 ratio less than store2(10/50=0.1<100/200=0.5)
+		bigger: NewStoreInfoWithAvailable(1, 10*gb, 100*gb, 1.4),
+		small:  NewStoreInfoWithAvailable(2, 100*gb, 1000*gb, 1.4),
+	}, {
+		// store1 and store2 has same usedSize and capacity,
+		// but the bigger's amp is bigger
+		bigger: NewStoreInfoWithAvailable(1, 10*gb, 100*gb, 1.5),
+		small:  NewStoreInfoWithAvailable(2, 10*gb, 100*gb, 1.4),
+	}, {
+		// store1 and store2 has same  capacity and regionSize（40g），
```
Comment (rleungx) [inline server/core/store_test.go:158]:
```suggestion
		// store1 and store2 has same capacity and regionSize (40g),

File: server/core/store_test.go [hunk 6]
```diff
@@ -127,3 +127,47 @@ func (s *testStoreSuite) TestLowSpaceRatio(c *C) {
...
+		// but store1 has less available space size
+		bigger: NewStoreInfoWithAvailable(1, 60*gb, 100*gb, 1),
+		small:  NewStoreInfoWithAvailable(2, 80*gb, 100*gb, 2),
+	}, {
+		// store1 and store2 has same  capacity and store2(40g) has twice usedSize than store1(20g)
+		// but store1 has higher amp ,so store1(60g) has more regionSize(40g)
```
Comment (rleungx) [inline server/core/store_test.go:164]:
```suggestion
		// but store1 has higher amp, so store1 (60g) has more regionSize (40g)
```

File: server/core/store_test.go [hunk 7]
```diff
@@ -127,3 +127,47 @@ func (s *testStoreSuite) TestLowSpaceRatio(c *C) {
...
+		// store1 and store2 has same available space and less than 50gb
+		bigger: NewStoreInfoWithAvailable(1, 10*gb, 1000*gb, 1.4),
+		small:  NewStoreInfoWithAvailable(2, 10*gb, 100*gb, 1.4),
+	}, {
+		// store1 and store2 has same available ratio less than 0.2
+		bigger: NewStoreInfoWithAvailable(1, 10*gb, 1000*gb, 1.4),
+		small:  NewStoreInfoWithAvailable(2, 1*gb, 100*gb, 1.4),
+	}, {
+		// store1 and store2 has same available ratio
+		// but the store1 ratio less than store2(10/50=0.1<100/200=0.5)
```
Comment (rleungx) [inline server/core/store_test.go:149]: 0.1?

Comment (bufferflies) [reply server/core/store_test.go:149]: the 0.1 is the radio : F-A/F and the F is max(capacity*(1-low_space_ratio),50).
so the less the ratio, the higher the score .

**Review by HunDunDM — APPROVED** [review]

**Review by nolouch — APPROVED** [review]
lgtm

**Pull request merged**